nemo/collections/asr/parts/submodules/multi_head_attention.py

```diff
@@ -315,19 +296,15 @@ def forward(self, x):
             x (torch.Tensor): Its shape is (batch, time, feature_size)
             pos_emb (torch.Tensor): Its shape is (1, time, feature_size)
         """
-        self.extend_pe(length=x.size(1))
-        if self.pe.dtype != x.dtype or self.pe.device != x.device:
-            self.pe = self.pe.to(device=x.device, dtype=x.dtype)
 
         if self.xscale:
             x = x * self.xscale
 
         # center_pos would be the index of position 0
         # negative positions would be used for right and positive for left tokens
         # for input of length L, 2*L-1 positions are needed, positions from (L-1) to -(L-1)
-        center_pos = self.pe.size(1) // 2
-        start_pos = center_pos - x.size(1) + 1
-        end_pos = center_pos + x.size(1)
+        start_pos = self.center_pos - x.size(1)
```

Comment:
Why center_pos is not calculated directly here?

Reply:
self.center_pos is precalculated in init()